src/client/kotlin/tech/thatgravyboat/skyblockapi/api/remote/hypixel/HypixelSkillAPI.kt

```diff
@@ -0,0 +1,81 @@
+package tech.thatgravyboat.skyblockapi.api.remote.hypixel
+
+import com.google.gson.JsonObject
+import kotlinx.coroutines.runBlocking
+import me.owdding.ktmodules.Module
+import tech.thatgravyboat.skyblockapi.api.remote.hypixel.HypixelSkillAPI.SkillData.Companion.toSkillData
+import tech.thatgravyboat.skyblockapi.utils.extentions.asInt
+import tech.thatgravyboat.skyblockapi.utils.extentions.asLong
+import tech.thatgravyboat.skyblockapi.utils.extentions.asString
+import tech.thatgravyboat.skyblockapi.utils.http.Http
+
+private const val API_URL = "https://api.hypixel.net/v2/resources/skyblock/skills"
+
+object HypixelSkillAPI {
+    enum class Skill : SkillType {
+        COMBAT,
+        FORAGING,
+        MINING,
+        FARMING {
+            override fun hasFloatingLevelCap(): Boolean = true
+        },
+        FISHING,
+        ENCHANTING,
+        ALCHEMY,
+        HUNTING,
+        TAMING {
+            override fun hasFloatingLevelCap(): Boolean = true
+        },
+        CARPENTRY,
+        RUNECRAFTING,
+        SOCIAL,
+        ;
+
+        private var internalSkillData: SkillData? = null
+        override val data: SkillData get() = internalSkillData ?: throw UnsupportedOperationException("Internal skill data is not yet supported")
```

Comment:
this error makes no sense and id like away to know if it exists without an error being thrown